linode/helper/stringplanmodifiers/usestateforunknownif.go

```diff
@@ -0,0 +1,68 @@
+package stringplanmodifiers
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+)
+
+// UseStateForUnknownIfNotNull is a convenience wrapper to only use the state value
+// in place of unknown values in plans if its value is not null.
+func UseStateForUnknownIfNotNull() planmodifier.String {
+	return UseStateForUnknownIf(
+		func(ctx context.Context, request planmodifier.StringRequest) bool {
+			return !request.StateValue.IsNull()
+		},
+	)
+}
+
+type planModifierCondition func(context.Context, planmodifier.StringRequest) bool
+
+// UseStateForUnknownIf returns a plan modifier that will only use the state value
+// in place of an unknown value if the given condition is met.
+func UseStateForUnknownIf(condition planModifierCondition) planmodifier.String {
+	return useStateForUnknownIfModifier{
+		conditionFunc: condition,
+	}
+}
```

Comment:
I wanted to make this type-agnostic but couldn't figure out a good way to do it. If anyone has any ideas some feedback would be much appreciated! We can revisit this in the future if not.